### ISSUE TITLE
Pressing Esc while dragging a files should cancel creating a linked file

### ISSUE THREAD

**Issue by ryonakano** [elementary/files#1327]
## Prerequisites
- [x] I have searched open and closed issues for duplicates.

## Feature
**Is your feature request related to a problem? Please describe.**
Files creates a linked file when you drag a file in the same folder (or move that file if it's another folder). However, this behavior does not seem to cancelable once you begin dragging.

![2020-05-18 11 39 45 の画面録画](https://user-images.githubusercontent.com/26003928/82169399-57c61000-98fc-11ea-8d2e-2f1c9c303e49.gif)

I pressed Esc while dragging in the screencast above, but a linked file is created against my expected behavior―the behavior is canceled and no linked file is created.

**Describe the solution you'd like**
Pressing Esc while dragging a files should cancel creating a linked file (or moving the file).

**Existing work**
This feature is implemented in Explorer on Windows and I always feel it's useful when you happen to began dragging a wrong file.

**Comment by jeremypw** [comment]
I find that pressing `Esc` during a drag does cancel the drag correctly already.  You must keep the primary button held down while pressing the key though as releasing the button drops the item.  This is with a English(UK) keyboard layout with default settings.

I must admit I just drag the item out of the window and drop it somewhere inactive if a start an intended drag as it is difficult to press `Esc` and hold down the button with one hand.

**Comment by ryonakano** [comment]
Ah, I seem to face this issue because I use tap to click instead of the primary button. When you doing a copy with the primary button, canceling with Esc works. However, doing the same thing with tap to click does not allow us to cancel with Esc.

**Comment by jeremypw** [comment]
I find it still works with "tap-to-click" with "single-click" off (single click selects).   With "single-click" on, (single click activates" I am unable to initiate a drag anyway.

**Comment by ryonakano** [comment]
Hmm, it does not work for me when "tap-to-click" is on and "single-click" is off (might be a hardware-specific issue though).

> With "single-click" on, (single click activates" I am unable to initiate a drag using just the touchpad anyway - is there a way of doing that?

I guess you can do by double-tapping a file you want to copy.

**Comment by jeremypw** [comment]
>I guess you can do by double-tapping a file you want to copy.

Oh yes, not sure why I could not manage that before :¬{
For me, `Esc` works for that as well.

Any other non-default settings/input method?

**Comment by ryonakano** [comment]
I'm using IBus as an input method. Also, I use [Fusuma](https://github.com/iberianpig/fusuma) so that I can use multi-touch gestures OS-widely. These might affect the situation. I'll set up a fresh Hera in English on my computer and take a look.

**Comment by ryonakano** [comment]
I've tested on a fresh Hera but the issue still persists. It might be a hardware-dependent issue…

**Comment by jeremypw** [comment]
Yes, I recently traced a Videos issue to the presence of Intel graphics/driver - did not affect NVidia graphics. Fortunately that was solved by using the most recent driver.

**Comment by jeremypw** [comment]
Are you using one of the Japanese keyboard layouts?

**Comment by jeremypw** [comment]
I just tried it on a fresh Hera stable install and confirm that this does happen there.  It does not happen on my unstable install though.
